pkg/auth/providers/activedirectory/activedirectory_client.go

```diff
@@ -147,6 +165,42 @@ func (p *adProvider) RefetchGroupPrincipals(principalID string, secret string) (
 	return groupPrincipals, err
 }
 
+func getUsernameFromObjectGUID(lConn *ldapv3.Conn, config *v32.ActiveDirectoryConfig, username string) (string, error) {
+	uuid := strings.TrimPrefix(username, "objectGUID=")
+
+	parsedGUID, err := guid.Parse(uuid)
+	if err != nil {
+		return "", fmt.Errorf("cannot parsing UUID: %w", err)
```

Comment:
This is basically repeating the same message as `guid.Parse` returns.

How about `parsing objectGUID to get the username: %w` which would provide more context?

Reply:
Yep, it make sense!